### PR TITLE
Add tests for reduction with identity parameter

### PR DESCRIPTION
**Pull request by vasilytric**

### CODE REVIEW

File: tests/reduction/reduction_common.h [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        {"char", "signed char", "unsigned char", "short int",
+         "unsigned short int", "int", "unsigned int", "long int",
+         "unsigned long int", "float", "long long int",
+         "unsigned long long int"});
+
+/** @brief Returns expected value for testing
+ *  @tparam VariableT The type of the variable with which the test runs
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam BufferT The type of the buffer with which the test runs
+ *  @param functor The functor (plus, multiplies e.t.c) with which the test runs
```
Comment (keryell) [inline tests/reduction/reduction_common.h:36]:
```suggestion
 *  @param functor The functor (plus, multiplies, etc.) with which the test runs
```

File: tests/reduction/reduction_common.h [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  auto buf_accessor =
+      buffer.template get_access<sycl::access_mode::read>(range);
+  for (int i{}; i < number_iterations; i++) {
+    expected_value = functor(buf_accessor[i], expected_value);
+  }
+  return expected_value;
+}
+
+/** @brief Filling buffer for using it in parallel_for and for calculate
+ * expected value
```
Comment (keryell) [inline tests/reduction/reduction_common.h:54]: Curious indent.

File: tests/reduction/reduction_get_lambda.h [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+#include <functional>
+
+namespace reduction_get_lambda {
+
+/** @brief Construct lambda for interacting with reducer while
+ *         sycl::handler.parallel_for used sycl::range
+ *  @tparam VariableT Variable type from type coverage
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam UseCombineFlagT std::integral_constant type that let switch between
+ *          calling .combine() function or operator +, *, ^= e.t.c.
```
Comment (keryell) [inline tests/reduction/reduction_get_lambda.h:22]:
```suggestion
 *          calling .combine() function or operator +, *, ^=, etc.

File: tests/reduction/reduction_get_lambda.h [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+    return [=](sycl::id<1> idx, auto &reducer) { reducer ^= accessor[idx]; };
+  }
+}
+
+/** @brief Construct lambda for interacting with reducer while
+ *         sycl::handler.parallel_for used sycl::nd_range
+ *  @tparam VariableT Variable type from type coverage
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam UseCombineFlagT std::integral_constant type that let switch between
+ *          calling .combine() function or operator +, *, ^= e.t.c.
```
Comment (keryell) [inline tests/reduction/reduction_get_lambda.h:58]:
```suggestion
 *          calling .combine() function or operator +, *, ^=, etc.
```

File: tests/reduction/reduction_get_lambda.h [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+  }
+}
+
+/** @brief Construct lambda for interacting with reducer in
+ *         sycl::handler.parallel_for
+ *  @tparam VariableT Variable type from type coverage
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam RangeT sycl::range or sycl::nd_range type
+ *  @tparam UseCombineFlagT std::integral_constant type that let switch between
+ *          calling .combine() function or operator +, *, ^= e.t.c.
```
Comment (keryell) [inline tests/reduction/reduction_get_lambda.h:104]:
```suggestion
 *          calling .combine() function or operator +, *, ^=, etc.
```

File: tests/reduction/reduction_get_lambda.h [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+        accessor);
+  }
+}
+
+/** @brief Construct lambda for interacting with reducer while
+ *         sycl::handler.parallel_for used sycl::range
+ *  @tparam VariableT Variable type from type coverage
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam UseCombineFlagT std::integral_constant type that let switch between
+ *          calling .combine() function or operator +, *, ^= e.t.c.
```
Comment (keryell) [inline tests/reduction/reduction_get_lambda.h:128]:
```suggestion
 *          calling .combine() function or operator +, *, ^=, etc.
```

File: tests/reduction/reduction_get_lambda.h [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+      }
+    };
+  }
+}
+
+/** @brief Construct lambda for interacting with reducer while
+ *         sycl::handler.parallel_for used sycl::nd_range
+ *  @tparam VariableT Variable type from type coverage
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam UseCombineFlagT std::integral_constant type that let switch between
```
Comment (keryell) [inline tests/reduction/reduction_get_lambda.h:185]:
```suggestion
 *  @tparam UseCombineFlagT std::integral_constant type that lets switch between
```

File: tests/reduction/reduction_get_lambda.h [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+ *          calling .combine() function or operator +, *, ^= e.t.c.
```
Comment (keryell) [inline tests/reduction/reduction_get_lambda.h:186]:
```suggestion
 *          calling .combine() function or operator +, *, ^=, etc.
```

File: tests/reduction/reduction_get_lambda.h [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+  }
+}
+
+/** @brief Construct lambda for interacting with reducer in
+ *         sycl::handler.parallel_for with using sycl::span
+ *  @tparam VariableT Variable type from type coverage
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam RangeT sycl::range or sycl::nd_range type
+ *  @tparam UseCombineFlagT std::integral_constant type that let switch between
+ *          calling .combine() function or operator +, *, ^= e.t.c.
```
Comment (keryell) [inline tests/reduction/reduction_get_lambda.h:245]:
```suggestion
 *  @tparam UseCombineFlagT std::integral_constant type that lets switch between
 *          calling .combine() function or operator +, *, ^=, etc.
```

File: tests/reduction/reduction_get_lambda.h [hunk 10]
```diff
@@ -0,0 +1,267 @@
...
+#include "reduction_common.h"
+#include <functional>
+
+namespace reduction_get_lambda {
+
+/** @brief Construct lambda for interacting with reducer while
+ *         sycl::handler.parallel_for used sycl::range
+ *  @tparam VariableT Variable type from type coverage
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam UseCombineFlagT std::integral_constant type that let switch between
```
Comment (keryell) [inline tests/reduction/reduction_get_lambda.h:21]:
```suggestion
 *  @tparam UseCombineFlagT std::integral_constant type that lets switch between
```

File: tests/reduction/reduction_get_lambda.h [hunk 11]
```diff
@@ -0,0 +1,267 @@
...
+                                    sycl::bit_xor<VariableT>>::value) {
+    return [=](sycl::id<1> idx, auto &reducer) { reducer ^= accessor[idx]; };
+  }
+}
+
+/** @brief Construct lambda for interacting with reducer while
+ *         sycl::handler.parallel_for used sycl::nd_range
+ *  @tparam VariableT Variable type from type coverage
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam UseCombineFlagT std::integral_constant type that let switch between
```
Comment (keryell) [inline tests/reduction/reduction_get_lambda.h:57]:
```suggestion
 *  @tparam UseCombineFlagT std::integral_constant type that lets switch between
```

File: tests/reduction/reduction_get_lambda.h [hunk 12]
```diff
@@ -0,0 +1,267 @@
...
+    };
+  }
+}
+
+/** @brief Construct lambda for interacting with reducer in
+ *         sycl::handler.parallel_for
+ *  @tparam VariableT Variable type from type coverage
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam RangeT sycl::range or sycl::nd_range type
+ *  @tparam UseCombineFlagT std::integral_constant type that let switch between
```
Comment (keryell) [inline tests/reduction/reduction_get_lambda.h:103]:
```suggestion
 *  @tparam UseCombineFlagT std::integral_constant type that lets switch between
```

File: tests/reduction/reduction_get_lambda.h [hunk 13]
```diff
@@ -0,0 +1,267 @@
...
+    return get_lambda_with_nd_range<VariableT, FunctorT, UseCombineFlagT>(
+        accessor);
+  }
+}
+
+/** @brief Construct lambda for interacting with reducer while
+ *         sycl::handler.parallel_for used sycl::range
+ *  @tparam VariableT Variable type from type coverage
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam UseCombineFlagT std::integral_constant type that let switch between
```
Comment (keryell) [inline tests/reduction/reduction_get_lambda.h:127]:
```suggestion
 *  @tparam UseCombineFlagT std::integral_constant type that lets switch between
```

File: tests/reduction/reduction_get_lambda.h [hunk 14]
```diff
@@ -0,0 +1,267 @@
...
+    };
+  }
+}
+
+/** @brief Construct lambda for interacting with reducer in
+ *         sycl::handler.parallel_for with using sycl::span
+ *  @tparam VariableT Variable type from type coverage
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam RangeT sycl::range or sycl::nd_range type
+ *  @tparam UseCombineFlagT std::integral_constant type that let switch between
```
Comment (keryell) [inline tests/reduction/reduction_get_lambda.h:244]:
```suggestion
 *  @tparam UseCombineFlagT std::integral_constant type that lets switch between
```

**Review by keryell — CHANGES_REQUESTED** [review]
That looks good.
Just a few typos.

**Review by ProGTX — APPROVED** [review]

File: tests/reduction/reduction_common.h (hunk 1 above)
Comment (vasilytric) [reply tests/reduction/reduction_common.h]: I've applied this comment in this commit: 212a7baae8e03f1861d99eb17467e7b054ca79f4

File: tests/reduction/reduction_common.h (hunk 2 above)
Comment (vasilytric) [reply tests/reduction/reduction_common.h]: Curious indent was fixed in this commit: ea039f0583fc00dfa57c74375de05863dae900d8

File: tests/reduction/reduction_get_lambda.h (hunk 3 above)
Comment (vasilytric) [reply tests/reduction/reduction_get_lambda.h]: I've applied this comment in this commit: 93364e17c64aebf054d2c7f722800ac0892bdc23

File: tests/reduction/reduction_get_lambda.h (hunk 4 above)
Comment (vasilytric) [reply tests/reduction/reduction_get_lambda.h]: I've applied this comment in this commit: 93364e17c64aebf054d2c7f722800ac0892bdc23

File: tests/reduction/reduction_get_lambda.h (hunk 5 above)
Comment (vasilytric) [reply tests/reduction/reduction_get_lambda.h]: I've applied this comment in this commit: 93364e17c64aebf054d2c7f722800ac0892bdc23

File: tests/reduction/reduction_get_lambda.h (hunk 14 above)
Comment (vasilytric) [reply tests/reduction/reduction_get_lambda.h]: I've applied this comment in this commit: 39139a01292d8b72557c33b4fa99352e6cf2d58f

File: tests/reduction/reduction_get_lambda.h (hunk 13 above)
Comment (vasilytric) [reply tests/reduction/reduction_get_lambda.h]: I've applied this comment in this commit: 39139a01292d8b72557c33b4fa99352e6cf2d58f

File: tests/reduction/reduction_get_lambda.h (hunk 12 above)
Comment (vasilytric) [reply tests/reduction/reduction_get_lambda.h]: I've applied this comment in this commit: 39139a01292d8b72557c33b4fa99352e6cf2d58f

File: tests/reduction/reduction_get_lambda.h (hunk 11 above)
Comment (vasilytric) [reply tests/reduction/reduction_get_lambda.h]: I've applied this comment in this commit: 39139a01292d8b72557c33b4fa99352e6cf2d58f

File: tests/reduction/reduction_get_lambda.h (hunk 10 above)
Comment (vasilytric) [reply tests/reduction/reduction_get_lambda.h]: I've applied this comment in this commit: 39139a01292d8b72557c33b4fa99352e6cf2d58f

File: tests/reduction/reduction_get_lambda.h (hunk 9 above)
Comment (vasilytric) [reply tests/reduction/reduction_get_lambda.h]: I've applied this comment in this two commits: 93364e17c64aebf054d2c7f722800ac0892bdc23 and 39139a01292d8b72557c33b4fa99352e6cf2d58f

File: tests/reduction/reduction_get_lambda.h (hunk 8 above)
Comment (vasilytric) [reply tests/reduction/reduction_get_lambda.h]: I've applied this comment in this commit: 93364e17c64aebf054d2c7f722800ac0892bdc23

File: tests/reduction/reduction_get_lambda.h (hunk 7 above)
Comment (vasilytric) [reply tests/reduction/reduction_get_lambda.h]: I've applied this comment in this commit: 93364e17c64aebf054d2c7f722800ac0892bdc23

File: tests/reduction/reduction_get_lambda.h (hunk 6 above)
Comment (vasilytric) [reply tests/reduction/reduction_get_lambda.h]: I've applied this comment in this commit: 93364e17c64aebf054d2c7f722800ac0892bdc23

**Review by keryell — APPROVED** [review]
Good work!

**Pull request merged**